wpe/src/main/glue/common/jnihelper.cpp

```diff
@@ -1,18 +1,39 @@
-#include <jnihelper.h>
-
-bool getJniEnv(JNIEnv **env) {
-    bool didAttachThread = false;
-    *env = nullptr;
-    auto get_env_result = vm->GetEnv((void**)env, JNI_VERSION_1_6);
-    if (get_env_result == JNI_EDETACHED) {
-        if (vm->AttachCurrentThread(env, NULL) == JNI_OK) {
-            didAttachThread = true;
+#include "jnihelper.h"
+
+#include <sys/prctl.h>
+
+namespace wpe {
+namespace android {
+namespace {
+
+JavaVM *g_jvm = NULL;
+
+}
+
+void InitVM(JavaVM *vm) {
+    g_jvm = vm;
+}
+
+JNIEnv *AttachCurrentThread() {
+    JNIEnv *env = nullptr;
+    jint ret = g_jvm->GetEnv(reinterpret_cast<void **>(&env), JNI_VERSION_1_6);
+    if (ret == JNI_EDETACHED || !env) {
+        JavaVMAttachArgs args;
```

Comment:
maybe initializing args here with `JavaVMAttachArgs args = {0};` would simplify code below because you will only need to set version and name if `(err >= 0)`

Moreover if tomorrow a version adds more members to the JavaVMAttachArgs struct they will be correctly initialized to 0 without having to change the code.

Reply:
Fixed